golem_sci/factory.py

```diff
@@ -64,11 +75,10 @@ def new_sci(
     _ensure_genesis(web3, chain)
     provider = ContractDataProvider(chain)
     geth_client = Client(web3)
-    nonce = geth_client.get_transaction_count(address)
     return SCIImplementation(
         geth_client,
         address,
-        JsonTransactionsStorage(datadir, nonce),
+        storage,
```

Comment:
Hi, @kbeker has noticed one problem with this change and I wanted to let you know.

The storage needs to get a nonce in its constructor. But to get the nonce the caller of `new_sci()` needs to access `geth_client` which is only created here.

Is it OK to create an instance of `Client(web3)` just to get the transaction count? Or would it be better to solve it differently?

Reply:
Yeah, this is something I'm thinking needs to change.
An alternative would be making constructors not to take nonce and introducing separate `init(nonce)` method that will be called in `SmartContractImplementation`'s constructor that will check for consistency. This way one does not need to provide nonce when creating the instance of TransactionsStorage.

Reply:
Yeah. That's probably the best we can do here.